libcnb/src/layer/handling.rs

```diff
@@ -107,6 +107,10 @@ fn handle_create_layer<B: Buildpack + ?Sized, L: Layer<Buildpack = B>>(
 ) -> Result<LayerData<L::Metadata>, HandleLayerErrorOrBuildpackError<B::Error>> {
     let layer_dir = context.layers_dir.join(layer_name.as_str());
 
+    fs::create_dir_all(&layer_dir)
```

Comment:
Actually, this is `layer_dir`, not `layers_dir`, so I don't know how this worked before for the Ruby example.

Reply:
@Malax Ah so it looks like this was indeed broken for buildpacks before (I've confirmed via adding an `.exists()` check to one of the examples), and the examples only worked because:
1. The hello world example doesn't write to the layer dir.
2. For the Ruby example:
   - the Ruby layer's first write to the `<layers>/ruby` layer dir is via the `tar` crate's `unpack()`, which [creates the directory](https://docs.rs/tar/0.4.37/src/tar/archive.rs.html#163-166).
   - the bundler layer's first write to the `<layers>/bundler` layer dir is via `bundle install --path <layer_dir>` which [creates the directory](https://github.com/rubygems/rubygems/blob/v3.2.33/bundler/lib/bundler/installer.rb#L69) if it doesn't exist. (The `gem install` command run prior to `bundle install` installs bundler into `<layers>/ruby` instead, so doesn't touch `<layers>/bundler`.)

So this fix is needed for more than just the tests :-)

Reply:
Thanks for looking into this before I could :) Yeah, I was puzzled too when I had to add this and thought that I'd investigate later why it works with the example buildpacks. 👍🏻